app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class LikesController < ApplicationController
+  def create
+    @user = Post.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @like = Like.new
+    @like.author = current_user
+    @like.post = @post
+
+    if @like.save
+      redirect_to user_post_path(@user, @post), notice: 'You have liked this post'
+    else
+      flash[:error] = @like.errors.full_messages
+      redirect_to user_post_path(@user, @post)
+    end
```

Comment:
- [ ] I think you should make use of the rails `respond_to` method for the same reason I indicated in my comment in your posts controller.
```ruby
respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
```